docs/architecture.md

```diff
@@ -3,46 +3,65 @@
 
 While we prefer in-code and live documentation over wiki pages, these are best suited to single services and we don't have a place to document broader processes that involve the interaction of several services. This section aims to provide that overview of how various services fit together.
 
-## Supporter data flow
+## `supporter-event-service`
+
+The service aims to propagate [supporter identities](https://github.com/comicrelief/data-models/tree/master/src/schemas/user-service/models/requests) to `serverless-user-service` and [supporter events](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions) to `serverless-erp-service`.
+
+`serverless-user-service` and `serverless-erp-service` exist on one region only because of their structure; should they be unavailable for any reason, propagation of data from upstream services may be interrupted. `supporter-event-service` is a multi-region service designed to handle incoming traffic from all sources. Events are queued to SQS and processed one by one.
+
+When identities are supplied to the `serverless-user-service` section of the payload, `supporter-event-service` returns a salted hash of the identity that can be used by upstream services to identity resources, so that anonymisation won't break relationships between data and services.
+
+`supporter-event-service` is supposed to handle only write requests to `serverless-user-service` and `serverless-erp-service`. If an upstream service requires data stored in either service, it should either:
+
+- **For critical paths**: store a copy of the data in the service itself
+- **For non-critical paths**: gracefully handle request failures
+
+Critical path here means whether it is acceptable or not to present a failure to and end user.
```

Comment:
Small typo: "and end user" -> "an end user".

Perhaps also rephrase this to explitly state what a critical path is: `A *critical path* is one where it is not acceptable to present a failure to an end user.`

Reply:
20a4a2bfb5271211e287ae50ed86cd6393710a28